src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -4231,53 +4231,41 @@ unsigned char * InstanceKlass::get_cached_class_file_bytes() {
 }
 #endif
 
-void InstanceKlass::log_to_classlist(const ClassFileStream* stream) const {
+bool InstanceKlass::is_shareable() const {
 #if INCLUDE_CDS
+  ClassLoaderData* loader_data = class_loader_data();
+  if (!SystemDictionaryShared::is_sharing_possible(loader_data)) {
+    return false;
+  }
+
+  if (is_hidden() || unsafe_anonymous_host() != NULL) {
+    tty->print_cr("skip writing %s class %s to classlist file",
+                  (is_hidden() ? "hidden" : "unsafe anonymous host"),
+                  name()->as_C_string());
+    return false;
+  }
+
+  if (module()->is_patched()) {
+    tty->print_cr("skip writing class %s to classlist file, module patched",
+                  name()->as_C_string());
+    return false;
+  }
+  return true;
+#else
+  return false;
+#endif
+}
+
+void InstanceKlass::log_to_classlist() const {
+#if INCLUDE_CDS
+  ResourceMark rm;
   if (ClassListWriter::is_enabled()) {
     if (!ClassLoader::has_jrt_entry()) {
        warning("DumpLoadedClassList and CDS are not supported in exploded build");
        DumpLoadedClassList = NULL;
        return;
     }
-    ClassLoaderData* loader_data = class_loader_data();
-    if (!SystemDictionaryShared::is_sharing_possible(loader_data)) {
-      return;
-    }
-    bool skip = false;
-    if (is_shared()) {
-      assert(stream == NULL, "shared class with stream");
-      if (is_hidden()) {
-        // Don't include archived lambda proxy class in the classlist.
-        assert(!is_non_strong_hidden(), "unexpected non-strong hidden class");
-        return;
-      }
-    } else {
-      assert(stream != NULL, "non-shared class without stream");
-      // skip hidden class and unsafe anonymous class.
-      if ( is_hidden() || unsafe_anonymous_host() != NULL) {
-        return;
-      }
-      oop class_loader = loader_data->class_loader();
-      if (class_loader == NULL || SystemDictionary::is_platform_class_loader(class_loader)) {
-        // For the boot and platform class loaders, skip classes that are not found in the
-        // java runtime image, such as those found in the --patch-module entries.
-        // These classes can't be loaded from the archive during runtime.
-        if (!stream->from_boot_loader_modules_image() && strncmp(stream->source(), "jrt:", 4) != 0) {
-          skip = true;
-        }
-
-        if (class_loader == NULL && ClassLoader::contains_append_entry(stream->source())) {
```

Comment:
Since the above has been removed, I think the `ClassLoader::contains_append_entry()` function can be removed too.